apps/studio/components/interfaces/Integrations/Queues/QueuesSettings.tsx

```diff
@@ -0,0 +1,251 @@
+    },
+  })
+
+  const { mutate: toggleExposeQueuePostgrest } = useDatabaseQueueToggleExposeMutation({
+    onSuccess: (_, values) => {
+      if (project && config) {
+        if (values.enable) {
+          const updatedSchemas = schemas.concat([QUEUES_SCHEMA])
+          updatePostgrestConfig({
+            projectRef: project?.ref,
+            dbSchema: updatedSchemas.join(', '),
+            maxRows: config.max_rows,
+            dbExtraSearchPath: config.db_extra_search_path,
+            dbPool: config.db_pool,
+          })
+        } else {
+          const updatedSchemas = schemas.filter((x) => x !== QUEUES_SCHEMA)
+          updatePostgrestConfig({
+            projectRef: project?.ref,
+            dbSchema: updatedSchemas.join(', '),
+            maxRows: config.max_rows,
+            dbExtraSearchPath: config.db_extra_search_path,
+            dbPool: config.db_pool,
+          })
+        }
+      }
+    },
+    onError: (error) => {
+      setIsToggling(false)
+      toast.error(`Failed to toggle queue exposure via PostgREST: ${error.message}`)
+    },
+  })
+
+  const onSubmit = async (values: z.infer<typeof formSchema>) => {
+    if (!project) return console.error('Project is required')
+    if (configError) {
+      return toast.error(
+        `Failed to toggle queue exposure via PostgREST: Unable to retrieve PostgREST configuration (${configError.message})`
+      )
+    }
+
+    setIsToggling(true)
+    toggleExposeQueuePostgrest({
+      projectRef: project.ref,
+      connectionString: project.connectionString,
+      enable: values.enable,
+    })
+  }
+
+  useEffect(() => {
+    if (isSuccess) form.reset({ enable: isExposed })
+  }, [isSuccess])
+
+  return (
+    <div className="w-full flex flex-col gap-y-4 p-10">
+      <FormHeader
+        className="mb-0"
+        title="Settings"
+        description="Manage your queues via any client library or PostgREST endpoints"
+      />
+      <Form_Shadcn_ {...form}>
+        <form id="pgmq-postgrest" onSubmit={form.handleSubmit(onSubmit)}>
+          <FormPanelContainer>
+            <FormPanelContent className="px-8 py-8">
+              <FormField_Shadcn_
+                control={form.control}
+                name="enable"
+                render={({ field }) => (
+                  <FormItem_Shadcn_ className="w-full">
+                    <FormItemLayout
+                      className="w-full"
+                      layout="flex"
+                      label="Expose PGMQ via PostgREST"
+                      description={
+                        <>
+                          <p>
+                            When enabled, you will be able to use any Supabase client library or
+                            PostgREST endpoints to manage your queues
+                          </p>
+                          <p className="mt-1">
+                            The following database functions will be available to use from the{' '}
```

Comment:
Tripped on the language here a little. 

"Use these functions to manage your queues: "  maybe?

Reply:
nit / opinion though ^^

Reply:
hmm agreed that this is a little odd, updated as such instead:
![image](https://github.com/user-attachments/assets/22c6f85a-49c9-43ad-ab41-eadb33aa0b26)